src/c++11/propagation.h

```diff
@@ -0,0 +1,93 @@
+// -*- Mode: C++ -*-
+#ifndef __LIGHTSTEP_PROPAGATION_H__
+#define __LIGHTSTEP_PROPAGATION_H__
+
+#include <functional>
+#include <string>
+
+namespace lightstep {
+  
+enum BuiltinFormat {
+  // Binary encodes the SpanContext for propagation as opaque binary data.
+  Binary = 1,  // NOT IMPLEMENTED
+
+  // TextMap encodes the SpanContext as key:value pairs.
+  TextMap = 2,  // NOT IMPLEMENTED
+
+  // HttpHeaders encodes the SpanContext as HTTP headers.
+  HttpHeaders = 3,
+};
+
+// Base class for implementation-provided or builtin carrier formats.
+class CarrierFormat {
+public:
+};
+
+// Builtin carrier format values.
+class BuiltinCarrierFormat : public CarrierFormat {
```

Comment:
why do we need both this and `BuiltinFormat`? (could the latter just be a `string`??)